dali/operators/reader/loader/webdataset_loader.cc

```diff
@@ -157,15 +157,12 @@ WebdatasetLoader::WebdatasetLoader(const OpSpec& spec)
     }
   }
 
-  auto dtypes_ids = spec.HasArgument("dtypes") ? spec.GetRepeatedArgument<DALIDataType>("dtypes") :
-                                                 std::vector<DALIDataType>(ext_.size(), DALI_UINT8);
-  dtypes_.reserve(dtypes_ids.size());
-  std::transform(dtypes_ids.begin(), dtypes_ids.end(), std::back_inserter(dtypes_),
-                 TypeTable::GetTypeInfo);
+  dtypes_ = spec.HasArgument("dtypes") ? spec.GetRepeatedArgument<DALIDataType>("dtypes")
+                                       : std::vector<DALIDataType>(ext_.size(), DALI_UINT8);
 
-  for (auto& dtype : dtypes_) {
-    DALI_ENFORCE(detail::wds::kSupportedTypes.count(dtype.id()),
-                 make_string("Unsupported output dtype ", dtype.name(),
+  for (DALIDataType dtype : dtypes_) {
+    DALI_ENFORCE(detail::wds::kSupportedTypes.count(dtype),
+                 make_string("Unsupported output dtype ", dtype,
```

Comment:
won't that get converted into an integer? Can you confirm that it produces a meaningful error?

Reply:
We do this everywhere, all the time.